barlayout.go

```diff
@@ -0,0 +1,145 @@
+package desktop
+
+import (
+	"math"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+)
+
+// Declare conformity with Layout interface
+var _ fyne.Layout = (*BarLayout)(nil)
+
+//BarLayout returns a layout used for zooming linear groups of icons
+type BarLayout struct {
```

Comment:
I don't think this type is needed outside of the bar, so don't export it (i.e. rename to barLayout).
Likewise the fields below are all private so there is no need to export them.